code/modules/mob/living/carbon/human/life.dm

```diff
@@ -269,12 +269,13 @@
 	return min(1,thermal_protection)
 
 /mob/living/carbon/human/handle_random_events()
-	//Puke if toxloss is too high
+	//Puke if toxloss or internal damage is too high
 	if(!stat)
-		if(getToxLoss() >= 45 && nutrition > 20)
+		if(getInternalLoss() >= 50 || (getToxLoss() >= 45 && nutrition > 20))
 			lastpuke += prob(50)
 			if(lastpuke >= 50) // about 25 second delay I guess
```

Comment:
Wouldn't this be an approximately once every 100 handle_random_events() ticks event, rather than a once every 25 seconds event?